### ISSUE TITLE
Can't turn with oculus touch controller.

### ISSUE THREAD

**Issue by stevejjones** [Codes4Fun/RBDOOM-3-BFG#49]
Game looks amazing and all controls are working fine except I can't turn using the thumb stick on the right hand Oculus touch controller.

I've tried vr_seated [0] vr_seated [1] vr_turning [0] vr_turning [1]

Strafing now works, but I had to use vr_strafing [0]

Thanks.

**Comment by Codes4Fun** [comment]
Unfortunately I do not have the Oculus Touch yet, but for motion controllers I have disabled turning with the thumb stick in favor of people physically turning. I might be able to re-enable this though, I have a working prototype that allows you to change weapons without a dedicated button.

vr_seated [1] has not been made to work with motion controllers, part of the problem being that they have so few buttons and you would need a crouch button, where as in standing mode, crouching in real life is crouching in game.

So I'm curious, are you able to do 360 turning with the Touch Controllers and their extra sensor, or would you just rather turn with the thumbstick?

**Comment by stevejjones** [comment]
I have been playing standing and I can do 360 turning with the Oculus touch and included extra sensor, but it would be better to also have the option to turn with the right stick as well, in-case you end up in a strange position at the edge of the play area and want to turn quickly. Just as strafing with the left stick is useful.

I know some people might get motion sickness with this, but I don't get any motion sickness from this in other games such as Windlands.

**Comment by Codes4Fun** [comment]
Actually, I take that back, vr_turning should work (but it doesn't work well, it needs to be tuned).

I'm actually confused on what exactly is happening. It sounds like for Touch for some reason vr_strafing is behaving opposite.

Are you using the SteamVR beta? Have you tried other SteamVR games/apps to see if the controls are working?

**Comment by stevejjones** [comment]
I'm not using the SteamVR beta. I'll download it and see if that fixes it.

**Comment by Codes4Fun** [comment]
I'm also kind of curious when you are in the SteamVR dashboard environment, what do the controllers look like (like Vive Wands?) and do the sticks appear to be working correctly in there?

**Comment by stevejjones** [comment]
I've tried the SteamVR beta. No change in Doom. When I'm in the SteamVR dashboard, I can see Vive Wands and I can see circles moving around the pads correctly as I move the analog sticks on the touch controllers.

**Comment by Codes4Fun** [comment]
That sucks, I guess we'll have to wait till tomorrow or Wednesday to find out what is going on, and for me to get a fix for it out there.

**Comment by stevejjones** [comment]
No problem. Touch isn't out until tomorrow anyway, so it's still really cool it works as well as it does. Thanks.

**Comment by stevejjones** [comment]
I tried RBDOOM-3-BFG 1.1.0 openvr X and strafing now works without adding vr_strafing [0]. Still can't get turning to work though.

**Comment by stevejjones** [comment]
Update: I tried release RBDOOM-3-BFG 1.1.0 openvr 4 and turning and strafing work perfectly with the oculus touch.

**Comment by Codes4Fun** [comment]
Out of curiosity what versions had you tried before?

**Comment by stevejjones** [comment]
I tried RBDOOM-3-BFG 1.1.0 openvr 7, RBDOOM-3-BFG 1.1.0 openvr 9, RBDOOM-3-BFG 1.1.0 openvr X and there were issues with strafing and turning.

RBDOOM-3-BFG 1.1.0 openvr 4 works perfectly.

**Comment by Codes4Fun** [comment]
I think maybe what is happening is that somehow the Touch controllers are being treated both as a gamepad and as Vive controllers.

If you ever try a later version again, try setting in_useJoystick [0], and see what happens, which either thumb sticks will do nothing or they will work but x axis will be inverted.

**Comment by Codes4Fun** [comment]
Also if you try that, do it in the game, not in the shell. That could affect the shell (I treat the touchpad as a d-pad which means to move up and down you have to click in the thumbstick), but I'm more interested in what happens with the control in the game.

**Comment by manefit** [comment]
it would be nice to have a seated with vr controller where the right stick does a 45 degree snap turn but u aim with the control still move forward and back as well as strafe with the left. I think this would this game amazing for those of us who don't wish to stand but still want to use motion control. even keeping the turn as is with the right stick but aim using motion control would be amazing. this might be hard to do so I won't get my hopes but I hope whoever is working on this mode could give that a try. if it works this could help show the way to other dev to do the same and maybe we'll get more ports of fps plus make it easier for devs to make vr fps

**Comment by baggyg** [comment]
Just thought I would let you know I have tested this with Touch today and the latest release (X) is working perfectly for me with a front facing setup. 
vr_seated 0
vr_turning 1 

Right stick does smooth turning (rather than snap). Motion controls working fine. The in_useJoystick change does not seem to be needed - it works on both settings. 

Just perfect. Let me know if there is something else you need me to test here

**Comment by manefit** [comment]
not getting the motion control on the touch to work

**Comment by baggyg** [comment]
@manefit Probably a little bit more information needed to be able to help... What config settings do you have? Generally if motion controllers are not working you may have vr_seated 1. Try changing this to vr_seated 0. Have you also been through the room setup on SteamVR?

**Comment by beezley** [comment]
motion control with touch not working for me either and they were before, maybe it has something to do with the steam update for the oculus controllers? vr_seated 0 and done room setup in steamvr but no dice,
hopefully can get a fix. fantastic mod!!! thank you dev 👍

**Comment by Codes4Fun** [comment]
I was thinking that for some reason the touch controllers are also sending gamepad input for games it thought didn't support motion controls.

Maybe try combination of starting SteamVR first, or not starting it first, or playing another game in SteamVR and then playing D3BFG.

I'd do this kind of testing myself, but I am still waiting on my Oculus Touch controllers, I have not even received the email yet. I missed ordering them on the 10th and ordered them 3 days after pre-orders were open, so I'm not sure how long it's going to take.

**Comment by baggyg** [comment]
I have the latest SteamVR Beta with the Touch models, so I'll have check tonight, but this shouldn't have broken anything. @beezley did you try "ini_useJoystick 0"?

**Comment by Codes4Fun** [comment]
Also check vr_moveMode, make sure it is 8. I imagine if you install a mod package from someone else they may have changed the default move mode, if that is the case, clicking in the stick might be how you have to move or just make sure vr_moveMode is at it's default.

**Comment by beezley** [comment]
Sorry Codes4Fun my bad, I had recently started using voice attack to boot games and the parameter in there was vr_seated +1. I removed that and it works fine, Sorry again for the confusion and my idiocy, thank you once again for a great mod :-) hope you get your controllers soon they are a work of art like your mod 👍

**Comment by manefit** [comment]
where is I find a break down of the console codes. the motion now works on the right hand but the turning is a little messed up it doesn't always work but when it does I love it also my it the button setup is weird and the a and x button does nothing. thanks for all ur hard work btw

**Comment by manefit** [comment]
i have vr_seated 0 vr_turning 1

**Comment by Codes4Fun** [comment]
There is a CONSOLE.TXT file in the zip that shows the ones I have added and ones from RB mod. There is also a CONTROLS.TXT that describes the buttons (although the flashlight is wrong, it is the left grip), and when you use vr_turning 1, it changes the right stick behavior.

I added also body slots recently, if you move your right hand over your right shoulder (like reaching in a backpack) and click the grip button it will get the next weapon, you can press it several times to cycle weapons. Similarly if you put your right arm behind your lower back it will get the previous weapon when you press the grip button. And for the PDA it is on your left hip, moving your left hand over it and pushing the grip button will bring up the PDA screen, you can also holster your current weapon on your right hip.

**Comment by manefit** [comment]
nice I will check it out thanks. just wondering cause I play seated with motion controllers. once u get your touch please have a seated with motion control option where u turn using the right stick  I figure it's cause u guys hasn't updated it and it sees the touch as both a gamepad and vr control is why I can play as I want to I just hope instead of removing it u guy refine it so it works smoother and just add is as an option. windlands is a great example of what I mean give it a try I think it will help you guys make a fps game everyone can love to play those like me who just don't wish to stand to play video games as well as those in WHEEL CHAIRS who CAN'T stand, duck and dodge...

**Comment by manefit** [comment]
I went back to RBDOOM-3-BFG 1.1.0 openvr 4 and now after a little messing around all but for couching works great with the touch for seated vr control play with free motion turning using the right stick. thank you guys I am working on posting a video if anyone is interested in the control mapping I use to get this working great for me

**Comment by Codes4Fun** [comment]
One of the hardest things about this game is that it requires a lot of buttons:
jump, duck, run, flashlight, fire, reload, pda, pause, next weapon, prev weapon

Both the Vive and Touch have:
Triggers, Grips, Touchpad/Thumbstick buttons, Menu Buttons, (and a non-usable system button)

So the game requires 10 buttons (but can easily use more with quick selecting weapons), while the controllers only have 8 usable buttons.

This is why I originally developed seated mode as only playable with a gamepad and haven't spent as much time on getting it to work with motion controllers. But my goal is definitely to figure this out.

**Comment by Codes4Fun** [comment]
I've added a console command in the [latest release](https://github.com/Codes4Fun/RBDOOM-3-BFG/releases/tag/1.1.0-openvr11) that will give me some information about Touch controllers. It would be nice if some of you could run that command. You can enter in the console vr_logDevices and then manually write out what it says or you can type these in the console:

logFile 2
vr_logDevices
exit

These commands will generated the file %userprofile%\Saved Games\id Software\RBDOOM 3 BFG\base\qconsole.log and quickly exists, so you can then open that file, copy and share the contents here. I'm curious about this both for someone who has working controls and from someone who doesn't

I did get an email that my Touch controllers have been processed, but no shipping info :/

**Comment by manefit** [comment]
https://www.youtube.com/watch?v=-jmFVNPH1Uc&t=28s I played seated check out my vid i'n using RBDOOM-3-BFG 1.1.0 openvr 4 cause it works best for touch seated. as for the buttons if u map jump to the A button, reload to the X button, use to the Y button, pause to the menu button, light to the left stick press, sprint to the left trigger, duck to the right stick press, next weapon to the right grab button, perv. to the left grab button, and pda to the b button I think u cover all u need for a great seated motion control doom with the touch and o how can I turn off movepoint it sucks and messing me up cause I didn't know it was on and didn't know y I was moving where I didn't want to go... thank for all your work I hope this can be helpful it plays great seated moving and turning with the sticks these mapping will help out a lot too I think with this doom will be the best fps in vr and u modders have a great base to take to other shooters if you wish.

**Comment by stevejjones** [comment]
I'm away from home just now, but I'll try the new release and console command when I get back in a few days. Hope you get your touch controllers soon!

**Comment by Codes4Fun** [comment]
I didn't realize that steamvr dashboard button was changed to be mapped to an Touch controllers 'enter' button and A and X are free to be used for what ever you want, instead of being used as dashboard buttons. That means Touch has two more buttons than the Wands, so I probably can try mapping controls differently for Touch.

I still need someone to use vr_logDevices when they get a chance, so I can know how to detect Touch controllers, or just wait until I get mine probably sometime this week and I'll figure it then.

**Comment by manefit** [comment]
I can do the log but won't get a chance to till Tuesday i got to spend time with the gf today and too busy tomorrow but free Tuesday hope someone else does it for u before then if not i will. Also how do i disable movepoint it's messing me up i just want to go forward when i push forward on the left stick no matter where i am pointing with it

**Comment by Codes4Fun** [comment]
For version 4 there is not a way to change "move point" but in version 11 it can be done with:
vr_moveDirection 0

This will make it that pressing forward will move in the direction you are looking instead of the direction of your left hand.

**Comment by manefit** [comment]
Damn that suck cause that worse than moving where i am point is there a way it can sticks act just like a normal fps game like a cod but i can then aim using motion control. But if the answer is no and I'll be happy with moving and turning remain as smooth as 4 but better button mapping I'll just keep my off hand pointing forward. Can i move the flashlight from my hand to lighting where i look cause it's when i use the flashlight that the movement messes me up

**Comment by Codes4Fun** [comment]
> Damn that suck cause that worse than moving where i am point is there a way it can sticks act just like a normal fps game like a cod but i can then aim using motion control.

Do you mean that you want to move based upon the right hand direction instead of the off hand? I also have that in version 11 using vr_moveDirection 2

> I'll be happy with moving and turning remain as smooth as 4

I'm definitely planning to improve moving and turning, version 4 was based on the Razer Hydra which has joysticks, where as all versions afterwards were based on Vive Wands which have trackpads. So it is just a matter of treating input as joysticks instead of trackpads for Touch.

There were also several bug fixes and features added since 4.

> Can i move the flashlight from my hand to lighting where i look cause it's when i use the flashlight that the movement messes me up

Currently there is not a way to do that, but I do plan to add that in the future.

**Comment by manefit** [comment]
Do you mean that you want to move based upon the right hand direction instead of the off hand? no I was just think none of my hand or look but more like the character's body if it's facing front I move forward then I turn using the left stick just as an traditional first person shooter. but if u move the flash light to the head so it light where I look then I would solely keep my off hand facing forward so I move it that direction. u got so great things plan and i'm shock u been so open to feedback from a vr gamer who wishes to play seated with the touch control. but in doing this u'll make this game great for tose who can't stand. on my youtube page a guy in a wheel chair reach out to me saying he can't play most of the game he wish to play cause dev focus on those who want to duck, dodge, and turn around and he can't do any of those thing thanks to u I was able to send him to try this all with setting I hope gives him a great time. so I wanted to share that to you cause while yes it how I wish to game but I saw that if I didn't post and recommend games that worked well seated I would have never think about those who can't stand but wish to play vr games too...

**Comment by Codes4Fun** [comment]
Oh, I get it now, forward should be a fixed direction of where you are seated, that is how vr_seated 1 works but doesn't work with motion controllers (yet). I'm moved this off into it's own issue #55 to get seated mode working, if you have other input regarding seated mode that would be the new place to leave it, and also subscribe to that issue (on the lower right) so when I fix it, you will be notified.

**Comment by Codes4Fun** [comment]
One-shot over in the Oculus Forums gave me a clue as to maybe what is going on, and I think I may have a fix for those having trouble. I updated the latest release with it.

version 12.2 https://github.com/Codes4Fun/RBDOOM-3-BFG/releases/tag/1.1.0-openvr12

I will not have my Touch controller until Friday, so I am not sure this will fix it but might be worth a try.

**Comment by stevejjones** [comment]
I've tried version 12 and it works great with oculus touch. I can turn fine with right stick using vr_rightaxis 1. I really like the ability to change weapons putting the controller over you shoulder. Thanks for all the hard work.
